ejb/src/main/java/org/wildfly/httpclient/ejb/RequestBuilder.java

```diff
@@ -0,0 +1,220 @@
+/*
+ * JBoss, Home of Professional Open Source.
+ * Copyright 2017 Red Hat, Inc., and individual contributors
+ * as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.wildfly.httpclient.ejb;
+
+import static io.undertow.util.Headers.ACCEPT;
+import static io.undertow.util.Headers.ACCEPT_ENCODING;
+import static io.undertow.util.Headers.CONTENT_ENCODING;
+import static io.undertow.util.Headers.CONTENT_TYPE;
+import static io.undertow.util.Headers.CHUNKED;
+import static io.undertow.util.Headers.GZIP;
+import static io.undertow.util.Headers.TRANSFER_ENCODING;
+
+import static org.wildfly.httpclient.ejb.EjbConstants.EJB_DISCOVERY_RESPONSE;
+import static org.wildfly.httpclient.ejb.EjbConstants.EJB_EXCEPTION;
+import static org.wildfly.httpclient.ejb.EjbConstants.INVOCATION_ACCEPT;
+import static org.wildfly.httpclient.ejb.EjbConstants.INVOCATION_ID;
+import static org.wildfly.httpclient.ejb.EjbConstants.INVOCATION;
+import static org.wildfly.httpclient.ejb.EjbConstants.SESSION_OPEN;
+import static org.wildfly.httpclient.ejb.RequestType.CANCEL_INVOCATION;
+import static org.wildfly.httpclient.ejb.RequestType.CREATE_SESSION;
+import static org.wildfly.httpclient.ejb.RequestType.DISCOVER;
+import static org.wildfly.httpclient.ejb.RequestType.START_INVOCATION;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+import static java.net.URLEncoder.encode;
+
+import io.undertow.client.ClientRequest;
+import io.undertow.util.HeaderMap;
+import org.jboss.ejb.client.EJBLocator;
+import org.wildfly.httpclient.common.Protocol;
+
+import java.lang.reflect.Method;
+
```

Comment:
Description of class and its purpose? Sample usage example would be nice as well, as certain methods need to be invoked before others (e.g. setting the request type).